site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,302 @@
+> 
+> - Your credentials: `wiener:peter`
+> - Victim's username: `carlos`
+> - [Candidate passwords](https://portswigger.net/web-security/authentication/auth-lab-passwords)
+
+Brute-force protection is especially important for applications that rely solely on password-based authentication. Without additional guard rails like multi-factor authentication (MFA), an attacker who can make multiple or unlimited guesses of users’ login credentials could eventually gain access to their accounts.
+
+However, even the best defense mechanisms can be undermined by poor implementation. If developers make incorrect assumptions about how users will interact with their application, they may unintentionally create vulnerabilities that attackers can exploit. For example, an attacker could manipulate normal application functionality to achieve results the developer neither intended nor expected.
+
+In this lab, we’ll use manipulate normal application functionality to access another user’s account. The lab provides us with a specific username, “carlos”, so we do not need to guess a valid username — we can focus on brute-forcing the password.
+
+Visit [the lab page](https://portswigger.net/web-security/authentication/password-based/lab-broken-bruteforce-protection-ip-block) and click the “Access the Lab” button to launch the test site. The URL for the lab will look something like “[https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/](https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/)”. You’ll want to copy it.
+
+For eagle-eyed readers, you may notice that the URLs in the screenshots vary slightly. This is because this article was written using multiple lab instances:
+
+&nbsp;
+![lab-home.png](images/lab-home.png)
+&nbsp;
+
+Now, let’s open the lab URL we just copied via ZAP. Click *Manual Explore* in the *Quick Start* tab:
+
+&nbsp;
+![quick-start-manual-explore-screen.png](images/quick-start-manual-explore-screen.png)
+&nbsp;
+
+Copy-paste the lab URL into the *URL to explore* field and click *Launch Browser*. Alternatively, launch your browser using the browser icon in the main toolbar:
+
+&nbsp;
+![launch-test-site-ZAP-configured-browser.png](images/launch-test-site-ZAP-configured-browser.png)
+&nbsp;
+
+> [!NOTE]
+> If you have any issues launching your browser from ZAP, see the following pages for help debugging the problem. Note that ZAP uses add-ons to enhance its core features, and one of these is the Selenium add-on, which provides drivers for interfacing with supported browsers. It is installed by default and exposes some configurable options:
+>
+> 1. [How can I fix 'browser was not found'?](https://www.zaproxy.org/faq/how-can-i-fix-browser-was-not-found/) - Zaproxy Docs (FAQ)
+> 2. [Manual Explore](https://www.zaproxy.org/docs/desktop/addons/quick-start/#manual-explore) - Zaproxy Docs
+> 3. [Selenium](https://www.zaproxy.org/docs/desktop/addons/selenium/) - Zaproxy Docs
```

Comment:
I guess you didn't test this?

![image](https://github.com/user-attachments/assets/c9c879eb-b245-4884-99dc-424f69bcb233)

Reply:
The callouts in the doc depends on changes added in #2977. I tested that PR with some sample content, but this PR doesn't contain those changes yet.